src/data/data_fetcher.py

```diff
@@ -273,3 +276,30 @@ def get_transactions_on_page(self):
         self.last_contribution_receipt_date = self.info["pagination"]["last_indexes"][
             "last_contribution_receipt_date"
         ]
+
+    def _build_df(self):
+        self.df = pd.DataFrame(
+            self.complete_list,
+            columns=[
+                "contributor_occupation",
+                "contributor_employer",
+                "contributor_city",
+                "contributor_state",
+                "contributor_zip",
+                "party",
+            ],
+        )
+        self.df.fillna(value="", inplace=True)
+        return self.df
```

Comment:
This all looks good and straight-forward. Two things:

1. I like to add a `self.df = None` in my `__init__()` for any variables that don't get defined until later-- for someone reading the code, it's helpful to know all of the attributes that can get tacked onto the class, even if they're not there yet. If they're always in the `__init__` (even as `None`) , there are no surprises later.
2. As mentioned above, we generally don't want the user calling `_` functions, let's lose the `return` statement and add a note to `self.gimmie_data()` about how the user will be magically have data in `.df` after running the function